src/actions/commands/insert.ts

```diff
@@ -551,9 +551,18 @@ class CommandCtrlUInInsertMode extends BaseCommand {
   keys = ['<C-u>'];
 
   public async exec(position: Position, vimState: VimState): Promise<VimState> {
-    const start = position.isInLeadingWhitespace()
-      ? position.getLineBegin()
-      : position.getLineBeginRespectingIndent();
+    let start: Position;
+
+    if (
+      vimState.insertModeCursorCtrlUStopPosition?.line === position.line &&
+      vimState.insertModeCursorCtrlUStopPosition?.isBefore(position)
+    ) {
+      start = vimState.insertModeCursorCtrlUStopPosition;
+    } else {
+      start = position.isInLeadingWhitespace()
+        ? position.getLineBegin()
+        : position.getLineBeginRespectingIndent();
+    }
     await TextEditor.delete(new vscode.Range(start, position));
```

Comment:
Since you're changing this action, maybe you could also change it to make use of transformations like `deleteRange` instead of changing the TextEditor directly. I'm not entirely sure, but I think this is preferred. Maybe @J-Fields can give some input here?!

Reply:
For my education, what's the advantage of the proposed approach? I'm open to changing it as long as @J-Fields agrees.

Reply:
Currently it won't change much. But the idea I think is to not have changes to editor on the actions but instead on the 'modeHandler' through transformations. If later someone tries to add some kind of compressing of changes to macros like @J-Fields did to the 'DocumentContentChangeAction' it will be harder if the actions are doing the changes themselves, instead of giving the transformation that needs to be done. But this is just my opinion.

Reply:
This design decision was made before I got involved, so I don't have the full story, but I think the idea is basically to have all the view changes in one place so the system as a whole is easier to reason about. It would probably make things like adopting VSCode's undo stack easier.

In theory I like the idea a lot, but I've found in practice it sometimes makes things *harder* to reason about, but I think that's mostly because `ModeHandler` is stuffed full of poorly documented accidental complexity.

So if it's simple to switch to a transformation, that'd be cool! If not, don't worry about it for now - once we have `ModeHandler` under more control, we can get to it (or possibly decide it was never a good idea in the first place and abandon it).

Reply:
Sounds good, I'll take a shot at using `deleteRange` instead